solo-chains/runtime/starlight/src/lib.rs

```diff
@@ -1606,6 +1611,76 @@ parameter_types! {
     pub const MaxLengthParaIds: u32 = 100u32;
     pub const MaxEncodedGenesisDataSize: u32 = 5_000_000u32; // 5MB
 }
+
+pub struct InnerStarlightRegistrar<Runtime, AccountId, RegistrarManager, RegistrarWeightInfo>(
+    PhantomData<(Runtime, AccountId, RegistrarManager, RegistrarWeightInfo)>,
+);
+impl<Runtime, AccountId, RegistrarManager, RegistrarWeightInfo> RegistrarHandler<AccountId>
+    for InnerStarlightRegistrar<Runtime, AccountId, RegistrarManager, RegistrarWeightInfo>
+where
+    RegistrarManager: RegistrarInterface<AccountId = AccountId>,
+    RegistrarWeightInfo: paras_registrar::WeightInfo,
+    Runtime: pallet_registrar::Config,
+{
+    fn register(
+        who: AccountId,
+        id: ParaId,
+        genesis_storage: Vec<ContainerChainGenesisDataItem>,
```

Comment:
Actually if this method doesn't need the full genesis storage, it would be much better to only take the code as input.

```suggestion
        code: &[u8],
```

Reply:
Yeah that would be nice, but the problem is that we need to take the code from that storage anyway, either inside this function or before calling it (inside `ContainerRegistrar::do_register` function). 

In my opinion is better to pass the genesis storage as a parameter and then search for the code inside the function. This way, we will perform this search only when `InnerRegistrar` has actually an implementation. Otherwise we would perform the search without much sense (like in Dancebox or Flashbox for instance).

Another solution is to receive the code as a parameter as we did with `head_data`, but I don't see any extra value on that, given that it's much more breaking and without much sense, as we already have the code inside `ContainerChainGenesisData`. What do you think?

Reply:
Great, yes at least we can avoid cloning the storage in that section. 

Mock fixed in e83f4a4c817cd829c039cb983531875273aa9f6b.